src/app/cards/add/Form.tsx

```diff
@@ -0,0 +1,293 @@
+        const reader = new FileReader();
+        convertImgToBase64(
+            reader,
+            file,
+            ({ resizedImgStr, error }: { resizedImgStr?: string; error?: string }) => {
+                if (!resizedImgStr) {
+                    console.log(error);
+                    setCard({ ...card, image_front: '' });
+                    return;
+                }
+                setCard({ ...card, image_front: resizedImgStr });
+            }
+        );
+    }
+
+    function handleImgBackChange(event: ChangeEvent<HTMLInputElement>) {
+        const file = event.target.files && event.target.files[0];
+        if (!file) {
+            setCard({ ...card, image_back: '' });
+            return;
+        }
+        const reader = new FileReader();
+        convertImgToBase64(
+            reader,
+            file,
+            ({ resizedImgStr, error }: { resizedImgStr?: string; error?: string }) => {
+                if (!resizedImgStr) {
+                    console.log(error);
+                    setCard({ ...card, image_back: '' });
+                    return;
+                }
+                setCard({ ...card, image_back: resizedImgStr });
+            }
+        );
+    }
+
+    async function addBtnClickHandler(event: React.FormEvent) {
+        event.preventDefault();
+        setLoading(true);
+
+        try {
+            const { id, error } = await handleCreateCard(new CardAdaptor(card));
+
+            setLoading(false);
+            if (id) {
+                router.replace(`/cards/${id}`);
+            }
+            if (error) {
+                setErrorMsg(error);
+            }
+            return;
+        } catch (error) {
+            setLoading(false);
+            setErrorMsg('Failed to add card');
+        }
+    }
+
+    function nextBtnClickHandler() {
+        if (progress === 0) {
+            if (!card.image_front) {
+                setErrorMsg('Please upload a photo.');
+                return;
+            }
+        }
+        setProgress((prevProgress) => prevProgress + 1);
+        setErrorMsg('');
+    }
+
+    function backBtnClickHandler() {
+        if (!progress) {
+            router.replace('/');
+        } else {
+            setProgress((prevProgress) => prevProgress - 1);
+        }
+    }
+
+    function getRightBtn() {
+        if (progress < 2) {
+            return (
+                <div onClick={nextBtnClickHandler} className={'text-info-600'}>
+                    Next
+                </div>
+            );
+        } else {
+            return (
+                <input
+                    type="submit"
+                    className={`${loading ? 'text-primary-600' : 'text-info-600'}`}
+                    disabled={loading}
+                    value={loading ? 'loading...' : 'Add'}
+                />
+            );
+        }
+    }
+
+    function getLeftBtn() {
+        return <div onClick={backBtnClickHandler}>Back</div>;
+    }
+
+    let inputfields;
```

Comment:
[NIT] I think it will look more neat if using an array to represent the inputfields(pages)

Reply:
Thank you